packages/bpk-docs/src/components/Header/Header.js

```diff
@@ -56,68 +56,90 @@ const toNavList = (links, hideOnTablet) => {
   }
 
   return (
-    <ul className={classNames.join(' ')}>
+    <BpkHorizontalNav className={classNames.join(' ')}>
       {links.map((link = {}) => (
-        <li
-          key={link.to || link.href}
-          className={getClassName('bpkdocs-header__nav-list-item')}
-        >
-          {link.to ? (
-            <BpkRouterLink
-              className={getClassName('bpkdocs-header__nav-list-link')}
-              {...link}
-            />
-          ) : (
-            <BpkLink
-              className={getClassName('bpkdocs-header__nav-list-link')}
-              {...link}
-            />
-          )}
-        </li>
+        <NavItemWithRouter key={link.to || link.href} link={link} />
       ))}
-    </ul>
+    </BpkHorizontalNav>
   );
 };
 
+const NavItemWithRouter = withRouter(props => {
+  const { link, router } = props;
+  const selected = link.to && router.isActive(link.to);
+  const itemProps = { selected };
+  if (link.to) {
+    // Fore BpkHorizontalNavItem to render an anchor.
+    // We use router.push to navigate below instead.
+    itemProps.href = '#';
+    itemProps.onClick = e => {
+      e.preventDefault();
+      router.push(link.to);
+    };
+    itemProps.className = getClassName(
+      'bpkdocs-header__horizontal-nav__nav-list-link',
+    );
+  } else {
+    itemProps.href = link.href;
+    itemProps.target = link.blank ? '_blank' : '_self';
+    itemProps.className = getClassName(
+      'bpkdocs-header__horizontal-nav__nav-list-link',
+    );
```

Comment:
(sorry for the repeat comment, I added it in the wrong place first)

This looks like it's repeated in both the `if` and `else`, can we move it up?

Reply:
Yeah this code can definitely be refactored anyway, will do if we agree with the approach

Reply:
Ok cool - in that case I definitely agree with the approach :)